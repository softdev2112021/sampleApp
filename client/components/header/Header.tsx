import SearchForm from "./searchForm/SearchForm";
import DropdownProfile from "./dropdownProfile/DropdownProfile";

interface HeaderProps {
  brandName: string;
  userName: string;
  avatar: string;
  onSearchSubmit: any;
  onLogout: () => void;
}

const Header = (props: HeaderProps) => {
  const { brandName, onSearchSubmit, ...dropdownProfileProps } = props;
  
  return (
    <div className="header navbar-default">
      <div className="navbar-brand">
        <span className="navbar-logo">
          <i className="fas fa-cloud"></i>
        </span>
        <b>{brandName}</b>&nbsp;App
      </div>
      <SearchForm onSubmit={onSearchSubmit} />
      <ul className="navbar-nav">
        <DropdownProfile {...dropdownProfileProps}/>
      </ul>
    </div>
  );
};

export default Header;
