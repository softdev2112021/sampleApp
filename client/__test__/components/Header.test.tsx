import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../../components/header/Header';

it('renders correctly', () => {
  const headerProps = {
    brandName: 'Forecastic',
    userName: 'username1',
    avatar: 'avatar',
    onSearchSubmit: () => {},
    onLogout: () => {},
  }

  const tree = renderer
    .create(<Header {...headerProps}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});