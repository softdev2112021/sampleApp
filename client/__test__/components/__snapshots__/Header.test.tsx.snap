// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="header navbar-default"
>
  <div
    className="navbar-brand"
  >
    <span
      className="navbar-logo"
    >
      <i
        className="fas fa-cloud"
      />
    </span>
    <b>
      Forecastic
    </b>
    Â App
  </div>
  <div
    className="navbar-form"
  >
    <form
      action=""
      method="POST"
      name="search_form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <div
          className="rbt"
          style={
            Object {
              "outline": "none",
              "position": "relative",
            }
          }
          tabIndex={-1}
        >
          <div
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "height": "100%",
                "position": "relative",
              }
            }
          >
            <input
              aria-autocomplete="both"
              aria-expanded={false}
              aria-haspopup="listbox"
              autoComplete="off"
              className="rbt-input-main form-control rbt-input"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Enter city name"
              role="combobox"
              type="text"
              value=""
            />
            <input
              aria-hidden={true}
              className="rbt-input-hint"
              readOnly={true}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "transparent",
                  "boxShadow": "none",
                  "color": "rgba(0, 0, 0, 0.35)",
                  "left": 0,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
              tabIndex={-1}
              value=""
            />
          </div>
        </div>
        <button
          className="btn btn-search"
          type="submit"
        >
          <i
            className="fa fa-search"
          />
        </button>
      </div>
    </form>
  </div>
  <ul
    className="navbar-nav"
  >
    <li
      className="dropdown navbar-user dropdown"
      onKeyDown={[Function]}
    >
      <a
        aria-expanded={false}
        aria-haspopup={true}
        className=""
        onClick={[Function]}
      >
        <img
          alt="profile"
          src="avatar"
        />
        <span
          className="d-none d-md-inline"
        >
          username1 
        </span>
        <b
          className="caret"
        />
      </a>
      <ul
        aria-hidden={true}
        className="dropdown-menu dropdown-menu-right dropdown-menu"
        role="menu"
        tabIndex="-1"
      >
        <button
          className="dropdown-item"
          onClick={[Function]}
          role="menuitem"
          tabIndex="0"
          type="button"
        >
          Log Out
        </button>
      </ul>
    </li>
  </ul>
</div>
`;
